### PR TITLE
v2.0.0-rc.0 - Edge runtime support and more.

### PR DESCRIPTION
**Pull request by vijayasingam-paddle**
## 2.0.0-rc.0 - 2024-09-26

> **Breaking changes:** This version includes major improvements that introduce breaking changes. These are called out below.

### Added

- Added support for edge runtime.

### Changed

- **Breaking change:** Updated the minimum required Node.js version to v18.
- **Breaking change:** `WebhooksValidator.isValidSignature` now returns a promise. 
- Enabled conditional exports based on runtimes.
- Switched from `node-fetch` to native `fetch` API.

---

### CODE REVIEW

File: src/__tests__/mocks/resources/subscriptions.mock.ts [hunk 1]
```diff
@@ -4,9 +4,9 @@
  *  Changes may be overwritten as part of auto-generation.
  */
 
-import { type ISubscriptionPreviewResponse, ISubscriptionResponse } from '../../../types';
```
Comment (samcolby) [inline src/__tests__/mocks/resources/subscriptions.mock.ts:7]: You can't do type imports in deno then?

File: src/__tests__/notifications/webhooks-validator.edge.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+describe('Webhooks validator in edge runtime', () => {
+  beforeAll(() => {
+    Object.defineProperty(global, 'crypto', {
+      value: {
+        subtle: crypto.webcrypto.subtle,
+      },
+    });
+    EdgeRuntime.initialize();
+  });
+  test.each([
```
Comment (samcolby) [inline src/__tests__/notifications/webhooks-validator.edge.test.ts:16]: Absolutely not a blocker.  I find this so much easier to read and get proper types for the test cases too

https://github.com/PaddleHQ/dashboard/blob/1afca98e5c8b782f41ca1b490bae5eefde8a67b7/apps/one-dashboard/src/utils/abs-money.test.ts#L3

File: src/internal/providers/crypto/edge-crypto.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { type CryptoProvider } from './crypto-provider.js';
+
+const byteHexMapping = new Array(256);
+for (let i = 0; i < byteHexMapping.length; i++) {
+  byteHexMapping[i] = i.toString(16).padStart(2, '0');
+}
+
+export class EdgeCrypto implements CryptoProvider {
```
Comment (samcolby) [inline src/internal/providers/crypto/edge-crypto.ts:8]: Do you have any tests for this?

Comment (samcolby) [reply src/internal/providers/crypto/edge-crypto.ts:8]: Specifically the `computeHmac` function

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,17 +12,18 @@ When we make [non-breaking changes](https://developer.paddle.com/api-reference/a
...
 
-## 2.0.0-beta.0 - 2024-09-24
+## 2.0.0-beta.0 - 2024-09-25
 
 > **Breaking changes:** This version includes major improvements that introduce breaking changes. These are called out below.
 
 ### Added
 
-- Added support for Bun, deno and edge runtimes.
+- Added support for edge runtime.
```
Comment (samcolby) [inline CHANGELOG.md]: Did you mean to remove `deno` from here?

File: src/__tests__/mocks/resources/subscriptions.mock.ts (hunk 1 above)
Comment (vijayasingam-paddle) [reply src/__tests__/mocks/resources/subscriptions.mock.ts:7]: I don't think this is a deno runtime requirenent. There is a [lint rule](https://typescript-eslint.io/rules/consistent-type-imports/) which preferred adding `type` keyword if they were types. 

Somehow it stopped adding it now 🤷

**Review by samcolby — DISMISSED** [review]
:fire:

File: src/internal/providers/crypto/edge-crypto.ts (hunk 3 above)
Comment (vijayasingam-paddle) [reply src/internal/providers/crypto/edge-crypto.ts:8]: we do have a test to assert it is generating correct HMAC [here](https://github.com/PaddleHQ/paddle-node-sdk/blob/364d4dc5f3d814ed4514e9ec1733c8d8057d3edd/src/__tests__/notifications/webhooks-validator.edge.test.ts)

**Review by mikeymike — DISMISSED** [review]

**Review by mikeymike — APPROVED** [review]

**Pull request merged**